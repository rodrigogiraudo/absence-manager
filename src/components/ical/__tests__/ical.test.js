/* eslint-disable no-undef */
import { render, screen, cleanup } from '@testing-library/react';
import ICalLink from '../index';

const event = {
  admitterId: null,
  admitterNote: '',
  confirmedAt: new Date('2017-01-09T18:43:29.000+01:00'),
  createdAt: new Date('2017-01-09T17:45:47.000+01:00'),
  crewId: 352,
  endDate: new Date('2017-01-11'),
  id: 2634,
  memberNote: 'Nachmittag 0,5 Tage. Danke.',
  rejectedAt: null,
  startDate: new Date('2017-01-11'),
  type: 'vacation',
  userId: 649,
  member: {
    crewId: 352,
    id: 713,
    image: 'http://place-hoff.com/300/400',
    name: 'Ines',
    userId: 649
  }
};

describe('iCal', () => {
  afterEach(cleanup);
  it('Should render the ical link component', () => {
    const { container } = render(<ICalLink events={[event]}>Download</ICalLink>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    render(<ICalLink events={[event]} />);
  });

  it('renders with text', () => {
    render(<ICalLink events={[event]}>Download</ICalLink>);
    expect(screen.getByTestId('ical-link')).toHaveTextContent('Download');
  });
});
