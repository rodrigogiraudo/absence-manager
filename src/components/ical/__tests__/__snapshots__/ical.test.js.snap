// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iCal Should render the ical link component 1`] = `
<a
  data-testid="ical-link"
  download="absences.ics"
  href="data:text/calendar;charset=utf8,%0ABEGIN%3AVCALENDAR%0AVERSION%3A2.0%0APRODID%3A-//bobbin%20v0.1//NONSGML%20iCal%20Writer//EN%0ACALSCALE%3AGREGORIAN%0AMETHOD%3APUBLISH%0A%0ABEGIN%3AVEVENT%0ADTSTART%3ATue%20Jan%2010%202017%20210000%20GMT0300%20%28Argentina%20Standard%20Time%29%0ADTEND%3ATue%20Jan%2010%202017%20210000%20GMT0300%20%28Argentina%20Standard%20Time%29%0ADTSTAMP%3A20210129T194808137Z%7D%0AUID%3A162421628@crewmeister.com%0ACREATED%3AMon%20Jan%2009%202017%20134547%20GMT0300%20%28Argentina%20Standard%20Time%29%0ADESCRIPTION%3AThe%20member%20Ines%20has%20a%20planned%20absence%20due%20to%20vacation%20starting%20Tue%20Jan%2010%202017%2021%3A00%3A00%20GMT-0300%20%28Argentina%20Standard%20Time%29%20until%20Tue%20Jan%2010%202017%2021%3A00%3A00%20GMT-0300%20%28Argentina%20Standard%20Time%29%0ASEQUENCE%3A0%0ASTATUS%3ACONFIRMED%0ASUMMARY%3AOOO%20Ines%20is%20on%20vacation%0ATRANSP%3AOPAQUE%0AEND%3AVEVENT%0AEND%3AVCALENDAR%0A"
>
  Download
</a>
`;
